packages/eslint-plugin/tests/rules/prefer-optional-chain.test.ts

```diff
@@ -180,6 +203,17 @@ ruleTester.run('prefer-optional-chain', rule, {
     'foo && foo[bar as string] && foo[bar as string].baz;',
     'foo && foo[1 + 2] && foo[1 + 2].baz;',
     'foo && foo[typeof bar] && foo[typeof bar].baz;',
+    '!foo || !foo[bar as string] || !foo[bar as string].baz;',
+    '!foo || !foo[1 + 2] || !foo[1 + 2].baz;',
+    '!foo || !foo[typeof bar] || !foo[typeof bar].baz;',
+    // currently do not handle 'this' as the first part of a chain
```

Comment:
Why not?

Reply:
I guess it could be helpful if the rule did work with `this` at the start of the chain, I was just trying to test more cases and found it by accident

Reply:
I figured it's ok to add this comment for the case that did not work and was not covered as there is a similar comment a few lines above it: https://github.com/typescript-eslint/typescript-eslint/blob/befa01f0c8c9166748c491fad01d0d800a368bb9/packages/eslint-plugin/tests/rules/prefer-optional-chain.test.ts#L197-L198

Reply:
Yeah, this can be a good followup issue 👍